### PR TITLE
Move generic mailbox docs from classic to main mailboxes doc

### PR DESCRIPTION
**Pull request by raboof**
'Mailbox configuration examples' could still be converted, but let's
keep PR's small

refs #27223

### CODE REVIEW

**Review by johanandren — COMMENTED** [review]
I think there are more parts that needs to be adapted and tweaked here, would rather see that in one go than moving it to typed first and then potentially leaving it here unfinished.

**Comment by raboof** [comment]
> I think there are more parts that needs to be adapted and tweaked here, would rather see that in one go than moving it to typed first and then potentially leaving it here unfinished.

You are probably right, let's do #27494 first though as that's related

**Comment by johanandren** [comment]
Something went bananas with the merge/rebase

**Comment by raboof** [comment]
> Something went bananas with the merge/rebase

oops, fixed

File: akka-docs/src/main/paradox/typed/mailboxes.md [hunk 1]
```diff
@@ -25,6 +25,8 @@ For advanced use cases it is also possible to defer mailbox selection to config
 
 ## Selecting what mailbox is used
 
+### Selecting a Mailbox Type for an Actor
+
 To select mailbox for an actor use `MailboxSelector` to create a `Props` instance for spawning your actor:
```
Comment (helena) [inline akka-docs/src/main/paradox/typed/mailboxes.md]: Can we add:
```suggestion
To select a mailbox for an actor use `MailboxSelector` to create a `Props` instance for spawning your actor:
```

File: akka-docs/src/main/paradox/typed/mailboxes.md [hunk 2]
```diff
@@ -37,4 +39,180 @@ Java
...
 @@snip [MailboxDocSpec.scala](/akka-actor-typed-tests/src/test/resources/mailbox-config-sample.conf) { }
 
-For more details on advanced mailbox config and custom mailbox implementations, see @ref[Classic Mailboxes](../mailboxes.md#builtin-mailbox-implementations).
+### Requiring a Message Queue Type for a Dispatcher
+
+A dispatcher may also have a requirement for the mailbox type used by the
+actors running on it. An example is the @apidoc[BalancingDispatcher] which requires a
+message queue that is thread-safe for multiple concurrent consumers. Such a
+requirement is formulated within the dispatcher configuration section like
+this:
```
Comment (helena) [inline akka-docs/src/main/paradox/typed/mailboxes.md]: I'm not sure the , "like this" is needed

File: akka-docs/src/main/paradox/typed/mailboxes.md [hunk 3]
```diff
@@ -37,4 +39,180 @@ Java
...
+requirement—then actor creation will fail.
+
+### How the Mailbox Type is Selected
+
+When an actor is created, the `ActorRefProvider` first determines the
+dispatcher which will execute it. Then the mailbox is determined as follows:
+
+ 1. If the actor’s deployment configuration section contains a `mailbox` key
+then that names a configuration section describing the mailbox type to be
+used.
```
Comment (helena) [inline akka-docs/src/main/paradox/typed/mailboxes.md]: The way this is written is confusing to me, can it be clarified - the sentence structure?

File: akka-docs/src/main/paradox/typed/mailboxes.md [hunk 4]
```diff
@@ -37,4 +39,180 @@ Java
...
+ 2. If the actor’s `Props` contains a mailbox selection then that names a configuration section describing the
+mailbox type to be used (note that this needs to be an absolute config path, 
+for example `myapp.special-mailbox`, and is not nested inside the `akka` namespace).
```
Comment (helena) [inline akka-docs/src/main/paradox/typed/mailboxes.md]: A bit run-on, perhaps separate into two sentences.

File: akka-docs/src/main/paradox/typed/mailboxes.md [hunk 5]
```diff
@@ -37,4 +39,180 @@ Java
...
+that requirement will be used to determine the mailbox type to be used; if
+that fails then the dispatcher’s requirement—if any—will be tried instead.
+ 5. If the dispatcher requires a mailbox type as described above then the
+mapping for that requirement will be used to determine the mailbox type to
+be used.
+ 6. The default mailbox `akka.actor.default-mailbox` will be used.
+
+### Default Mailbox
+
+When the mailbox is not specified as described above the default mailbox
```
Comment (helena) [inline akka-docs/src/main/paradox/typed/mailboxes.md]:
```suggestion
The default mailbox is used when the mailbox is not specified.
```

File: akka-docs/src/main/paradox/typed/mailboxes.md [hunk 6]
```diff
@@ -37,4 +39,180 @@ Java
...
+is used. By default it is an unbounded mailbox, which is backed by a
```
Comment (helena) [inline akka-docs/src/main/paradox/typed/mailboxes.md]:
```suggestion
which is an unbounded mailbox, backed by a
```

File: akka-docs/src/main/paradox/typed/mailboxes.md [hunk 7]
```diff
@@ -37,4 +39,180 @@ Java
...
+  mailbox-type = "akka.dispatch.SingleConsumerOnlyUnboundedMailbox"
+}
+```
+
+### Which Configuration is passed to the Mailbox Type
+
+Each mailbox type is implemented by a class which extends `MailboxType`
+and takes two constructor arguments: a `ActorSystem.Settings` object and
+a `Config` section. The latter is computed by obtaining the named
+configuration section from the actor system’s configuration, overriding its
```
Comment (helena) [inline akka-docs/src/main/paradox/typed/mailboxes.md]:
```suggestion
configuration section from the `ActorSystem` configuration, overriding its
```

File: akka-docs/src/main/paradox/typed/mailboxes.md [hunk 8]
```diff
@@ -37,4 +39,180 @@ Java
...
+`id` key with the configuration path of the mailbox type and adding a
+fall-back to the default mailbox configuration section.
+
+## Builtin Mailbox Implementations
```
Comment (helena) [inline akka-docs/src/main/paradox/typed/mailboxes.md]:
```suggestion
## Mailbox Implementations
```

File: akka-docs/src/main/paradox/typed/mailboxes.md [hunk 9]
```diff
@@ -37,4 +39,180 @@ Java
...
+
+Akka comes shipped with a number of mailbox implementations:
```
Comment (helena) [inline akka-docs/src/main/paradox/typed/mailboxes.md]:
```suggestion
Akka shipps with a number of mailbox implementations:
```

File: akka-docs/src/main/paradox/typed/mailboxes.md [hunk 10]
```diff
@@ -37,4 +39,180 @@ Java
...
+ * 
+   **UnboundedMailbox** (default)
+    * The default mailbox
+    * Backed by a `java.util.concurrent.ConcurrentLinkedQueue`
+    * Blocking: No
+    * Bounded: No
+    * Configuration name: `"unbounded"` or `"akka.dispatch.UnboundedMailbox"`
+ * 
+   **SingleConsumerOnlyUnboundedMailbox**
+   This queue may or may not be faster than the default one depending on your use-case—be sure to benchmark properly!
```
Comment (helena) [inline akka-docs/src/main/paradox/typed/mailboxes.md]:
```suggestion
Depending on your use case, this queue may or may not be faster than the default, so you must be sure to benchmark properly.
```

File: akka-docs/src/main/paradox/typed/mailboxes.md [hunk 11]
```diff
@@ -37,4 +39,180 @@ Java
...
+ * **BoundedControlAwareMailbox**
+    * Delivers messages that extend `akka.dispatch.ControlMessage` with higher priority
+    * Backed by two `java.util.concurrent.ConcurrentLinkedQueue` and blocking on enqueue if capacity has been reached
+    * Blocking: Yes if used with non-zero `mailbox-push-timeout-time`, otherwise No
+    * Bounded: Yes
+    * Configuration name: "akka.dispatch.BoundedControlAwareMailbox"
+
+## Creating your own Mailbox type
+
+An example is worth a thousand quacks:
```
Comment (helena) [inline akka-docs/src/main/paradox/typed/mailboxes.md]:
```suggestion
The best way to show how to roll your own Mailbox type is by example:
```

File: akka-docs/src/main/paradox/typed/mailboxes.md [hunk 12]
```diff
@@ -37,4 +39,180 @@ Java
...
+    * Bounded: Yes
+    * Configuration name: "akka.dispatch.BoundedStablePriorityMailbox"
+ * **BoundedControlAwareMailbox**
+    * Delivers messages that extend `akka.dispatch.ControlMessage` with higher priority
+    * Backed by two `java.util.concurrent.ConcurrentLinkedQueue` and blocking on enqueue if capacity has been reached
+    * Blocking: Yes if used with non-zero `mailbox-push-timeout-time`, otherwise No
+    * Bounded: Yes
+    * Configuration name: "akka.dispatch.BoundedControlAwareMailbox"
+
+## Creating your own Mailbox type
```
Comment (helena) [inline akka-docs/src/main/paradox/typed/mailboxes.md]:
```suggestion
## Custom Mailbox type
```

**Review by helena — COMMENTED** [review]
Looking great! Some suggestions.

**Review by johanandren — APPROVED** [review]
LGTM

I'd like us to simplify this section even further and mention less shoot-yourself-in-the-foot options to not encourage newcomers to end up doing crazy stuff, but that is sort of out of scope for this PR.

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**